### ISSUE TITLE
ID of generateSitemaps doesn't work in production for sitemap

### ISSUE THREAD

**Issue by hc0503** [vercel/next.js#60894]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/adoring-spence-ypzmjh?file=%2Fapp%2Fbooks%2Fsitemap.ts%3A5%2C1

### To Reproduce

If I call `/books/sitemap/10.xml` in production, the console.log shows 1.
That should return `not found` or 10
This bug is happened in only production.

### Current vs. Expected behavior

That should return correct id. If it's existed, it should return `not found` like dev mode. 

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Home  
Binaries:
  Node: 20.9.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.1.0
  eslint-config-next: 14.1.0
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router

### Which stage(s) are affected? (Select all that apply)

next build (local)

### Additional context

The generateSitemaps works well in development mode, but that has above error in production mode.
It seems like be Next.js issue.

**Comment by craftedsro** [comment]
I have same issue.

**Comment by huozhi** [comment]
This fix is landed in 14.1.2
